app/services/optIn/OptInService.scala

```diff
@@ -165,4 +169,50 @@ class OptInService @Inject()(itsaStatusUpdateConnector: ITSAStatusUpdateConnecto
       case Some(s) => s.selectedOptInYear.flatMap(TaxYear.getTaxYearModel)
     }
   }
+
+  /* todo may need to be removed or changed, awaiting ongoing investigation and decisions by team */
+  def getQuarterlyUpdatesCountForOfferedYears(proposition: OptInProposition)
+                                             (implicit user: MtdItUser[_],
+                                              hc: HeaderCarrier,
+                                              ec: ExecutionContext): Future[QuarterlyUpdatesCountForTaxYearModel] = {
+
+    def cumulativeQuarterlyUpdateCounts(taxYearToCount: Seq[QuarterlyUpdatesCountForTaxYear]): Seq[QuarterlyUpdatesCountForTaxYear] =
+      if (taxYearToCount.isEmpty)
+        Seq()
+      else
+        Seq(cumulativeCount(taxYearToCount)) ++ cumulativeQuarterlyUpdateCounts(taxYearToCount.tail)
+
+    def cumulativeCount(taxYearToCount: Seq[QuarterlyUpdatesCountForTaxYear]): QuarterlyUpdatesCountForTaxYear =
+      QuarterlyUpdatesCountForTaxYear(taxYearToCount.head.taxYear, taxYearToCount.map(_.count).sum)
+
+    val annualQuarterlyUpdateCounts = Future.sequence(
+      proposition.availableOptInYears.map {
+        case next: NextOptInTaxYear => Future.successful(QuarterlyUpdatesCountForTaxYear(next.taxYear, ZeroCount))
+        case anyOtherOptInTaxYear => nextUpdatesService.getQuarterlyUpdatesCounts(anyOtherOptInTaxYear.taxYear)
+      })
+
+    annualQuarterlyUpdateCounts.
+      map(cumulativeQuarterlyUpdateCounts).
+      map(QuarterlyUpdatesCountForTaxYearModel)
+  }
+
+  def getMultiYearCheckYourAnswersViewModel(isAgent: Boolean)(implicit user: MtdItUser[_],
+                                                              hc: HeaderCarrier,
+                                                              ec: ExecutionContext): Future[Option[MultiYearCheckYourAnswersViewModel]] = {
+    val result = for {
+      intentTaxYear <- OptionT(fetchSavedChosenTaxYear())
+      cancelURL = routes.ReportingFrequencyPageController.show(isAgent).url
+      intentIsNextYear = isNextTaxYear(dateService.getCurrentTaxYear, intentTaxYear)
+    } yield MultiYearCheckYourAnswersViewModel(intentTaxYear, isAgent, cancelURL, intentIsNextYear)
+
+    result.value
+  }
+
+  private def isNextTaxYear(currentTaxYear: TaxYear, nextTaxYear: TaxYear): Boolean = currentTaxYear.nextYear == nextTaxYear
```

Comment:
A small one, but could the second parameter `nextTaxYear` be renamed `candidate` or something similar. We're testing if it is the next tax year. The way it is right now a reader of the code may assume it is definitely the next tax year.

Reply:
done